### DISCUSSION TITLE
List color per namespace

### DISCUSSION THREAD

**Discussion by ToshY** [go-task/task#824] [Ideas]
I recently switched from GNU Make to Taskfile, and after a while I realized that there's just one thing for me: colors.

Currently, when displaying available tasks with `--list`, it shows a list of all tasks prefixed per namespace in **white**. Think it would be nice to have an option to specify the color per namespace as this makes it easier to visually distinguish between them.

I'm not familiar with Taskfile's innerworkings, but I'd reckon specifying a (environment) variable color in each Taskfile, and then use that when displaying tasks, could work maybe (?)

Another possible (fun) idea would just to randomize the colors per namespace, giving a kind of 🌈 effect.

**Comment by ghostsquad** [comment]
similar to stern colorizing each pod (when multiple pods are found).

![image](https://user-images.githubusercontent.com/903488/182042289-afa21d77-a6ed-4b07-811a-1513e0033837.png)

Though, if this were implemented, it would make sense for the following:
- deterministic behavior: e.g. seeded random behavior (colors don't change between runs)
- allow a user to specify namespace attributes (e.g. manually specify color in hex format or simple strings, like "blue")
- allow color to be turned on/off, via cli flag and environment variable

Given that this isn't backwards compatible, the last one would be pretty important, so individual users can enable colors as _their_ default.

**Reply by andreynering** [reply]
If we decide to proceed with this, we should definitely make it deterministic.

**Reply by ToshY** [reply]
hey @andreynering 👋 

Is this still something that can be considered as a (low-priority) feature request?

**Reply by andreynering** [reply]
@ToshY I just opened an issue about it: https://github.com/go-task/task/issues/1668

**Comment by davidbarratt** [comment]
I'm coming from [concurrently](https://www.npmjs.com/package/concurrently) that has this feature and I was disappointed that Task doesn't. 😔
